### PR TITLE
add baisc auth

### PR DESCRIPTION
**Pull request by ayashbaral**
i. baisc auth for rest api enabled
ii. secret for auth added in secret yaml

### CODE REVIEW

File: todo-app/src/main/java/com/klab/todo/security/TodoWebSecurityConfig.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.csrf().disable().authorizeRequests().antMatchers("/actuator/info", "/actuator/health").permitAll()
+                .anyRequest().authenticated().and().httpBasic();
+
+    }
+
+    @Autowired
+    public void configureGlobal(AuthenticationManagerBuilder auth) throws Exception {
```
Comment (Raghav2211) [inline todo-app/src/main/java/com/klab/todo/security/TodoWebSecurityConfig.java:28]: 👍

File: todo-app/src/main/java/com/klab/todo/service/ITodoService.java [hunk 2]
```diff
@@ -10,10 +10,10 @@
 
     public Todo update(Todo user);
 
-    public Optional<Todo> findById(long id);
+    public Optional<Todo> findById(Long id);
```
Comment (Raghav2211) [inline todo-app/src/main/java/com/klab/todo/service/ITodoService.java:13]: 👍

File: todo-app/src/main/resources/application.properties [hunk 3]
```diff
@@ -19,5 +19,9 @@ management.endpoint.health.show-details=always
 # enable all actuator endpoints
 management.endpoints.web.exposure.include=*
  
-# build version
-klab.todo.build.version=@project.version@
+#basic auth security config details
+api.secuirity.username=${APP_USER}
+api.secuirity.password=${APP_PASSWORD}
```
Comment (Raghav2211) [inline todo-app/src/main/resources/application.properties:24]: ${AUTH_USER} , ${AUTH_PASSWORD} ?

File: todo-app/src/main/resources/application.properties [hunk 4]
```diff
@@ -19,5 +19,9 @@ management.endpoint.health.show-details=always
...
+
+
+server.error.include-message=always
```
Comment (Raghav2211) [inline todo-app/src/main/resources/application.properties:27]: 👍

File: todo-app/local-kube-env/todo-secret.yaml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: todo-secret
+type: Opaque
+data:
+  username: YWRtaW4=
+  password: e25vb3B9cGFzc3dvcmQ=
```
Comment (Raghav2211) [inline todo-app/local-kube-env/todo-secret.yaml:8]: convention same as application.properties

File: todo-app/docker-compose.yaml [hunk 6]
```diff
@@ -21,3 +21,5 @@ services:
       MYSQL_USER: root
       MYSQL_PASSWORD: root
       MYSQL_DB_NAME: klab
+      APP_USER: admin
+      APP_PASSWORD: "{noop}password"
```
Comment (Raghav2211) [inline todo-app/docker-compose.yaml:25]: convention same as application.properties

File: todo-app/local-kube-env/todo-secret.yaml (hunk 5 above)
Comment (ayashbaral) [reply todo-app/local-kube-env/todo-secret.yaml:8]: done

File: todo-app/docker-compose.yaml (hunk 6 above)
Comment (ayashbaral) [reply todo-app/docker-compose.yaml:25]: done

File: todo-app/src/main/resources/application.properties (hunk 3 above)
Comment (ayashbaral) [reply todo-app/src/main/resources/application.properties:24]: done

**Review by Raghav2211 — APPROVED** [review]

**Pull request merged**